contracts/facets/EmbalmerFacet.sol

```diff
@@ -60,24 +50,27 @@ contract EmbalmerFacet {
     /// finalizeSarcohpagus() method should be called, which is the second step.
     ///
     /// @param name the name of the sarcophagus
-    /// @param sarcoId the identifier of the sarcophagus
     /// @param archaeologists the data for the archaeologists
     /// @param arweaveArchaeologist The address of the archaeologist who uploads to arweave
     /// @param recipient the address of the recipient
     /// @param resurrectionTime the resurrection time of the sarcophagus
+    /// @param maxResurrectionInterval the maximum length of time that any new resurrection times can be from time of rewrap
+    /// @dev archaeologists will have to sign off on this interval, and commit to it for the lifetime of the sarcophagus.
     /// @param canBeTransferred Whether the sarcophagus can be transferred
     /// @param minShards The minimum number of shards required to unwrap the sarcophagus
     /// @return The index of the new sarcophagus
     function initializeSarcophagus(
         string memory name,
-        bytes32 sarcoId,
         LibTypes.ArchaeologistMemory[] memory archaeologists,
         address arweaveArchaeologist,
         address recipient,
         uint256 resurrectionTime,
+        uint256 maxResurrectionInterval,
         bool canBeTransferred,
         uint8 minShards
     ) external returns (uint256) {
+        bytes32 sarcoId = keccak256(abi.encodePacked(name));
```

Comment:
Seems the stack too deep error is fixed when the _position_ of the `sarcoId` param is last. 

Also there's some weird stuff going on with different permutations of the params, but it looks like the array of `archaeologists`, and its position relative to the `sarcoId`, influences whether it works or not. 

If both are (almost) last, it works, otherwise if they're too close to each other, _and_ above the `address` type params, then we get stack too deep.